packages/gatsby/src/redux/reducers/component-data-dependencies.ts

```diff
@@ -12,36 +17,36 @@ module.exports = (
 
       // If this nodeId not set yet.
       if (action.payload.nodeId) {
-        let existingPaths = new Set()
+        let existingPaths: Set<string> = new Set()
         if (state.nodes.has(action.payload.nodeId)) {
-          existingPaths = state.nodes.get(action.payload.nodeId)
+          existingPaths = state.nodes.get(action.payload.nodeId)!
         }
-        if (!existingPaths.has(action.payload.path || action.payload.id)) {
-          existingPaths.add(action.payload.path || action.payload.id)
+        if (!existingPaths.has(action.payload.path)) {
+          existingPaths.add(action.payload.path)
         }
         state.nodes.set(action.payload.nodeId, existingPaths)
       }
 
       // If this connection not set yet.
       if (action.payload.connection) {
-        let existingPaths = new Set()
+        let existingPaths: Set<string> = new Set()
         if (state.connections.has(action.payload.connection)) {
-          existingPaths = state.connections.get(action.payload.connection)
+          existingPaths = state.connections.get(action.payload.connection)!
```

Comment:
this is so weird that TS is not smart enough to know that this for sure exists given the check above it, but 🤷‍♂️